bazel_tools/client_server/client_server_build.bzl

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2019 The DAML Authors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def _client_server_build_impl(ctx):
+    ctx.actions.run_shell(
+        outputs = [ctx.outputs.out],
+        inputs = ctx.files.data,
+        tools = depset([
+            ctx.executable._runner,
+            ctx.executable.client,
+            ctx.executable.server,
+        ]),
+        use_default_shell_env = True,
+        command = """
+        export {output_env}="{output_path}"
+        {runner} "{client}" "{client_args} {client_files}" "{server}" "{server_args} {server_files}" &> runner.log
+        if [ "$?" -ne 0 ]; then
+          cat runner.log
+          exit 1
+        fi
+      """.format(
+            output_env = ctx.attr.output_env,
+            output_path = ctx.outputs.out.path,
+            runner = ctx.executable._runner.path,
+            client = ctx.executable.client.path,
+            server = ctx.executable.server.path,
+            server_args = " ".join(ctx.attr.server_args),
+            server_files = " ".join([f.path for f in ctx.files.server_files]),
+            client_args = " ".join(ctx.attr.client_args),
+            client_files = " ".join([f.path for f in ctx.files.client_files]),
+        ),
+    )
+    return [
+        DefaultInfo(
+            files = depset([ctx.outputs.out]),
+            runfiles = ctx.runfiles(files = ctx.files.data, collect_data = True),
```

Comment:
Are files passed in via `data` really needed in `runfiles`? As I understand they are files required by the `client` or `server` during the execution of the `runner` command.
```suggestion
```

Reply:
good point. wasn't needed.